### PR TITLE
#2v50h

### PR DESCRIPTION
**Pull request by jjkramok**

### CODE REVIEW

File: core/src/com/marama/view/View.java [hunk 1]
```diff
@@ -4,15 +4,35 @@
...
+import org.apache.commons.exec.ExecuteException;
+import java.io.IOException;
 
 /**
  * This is the initial {@link Game} class that will be instantiated by libGDX in the different launchers.
  */
 public class View extends Game {
     @Override
     public void create () {
+        // TODO move code to another class or method
```
Comment (wilbo) [inline core/src/com/marama/view/View.java]: This can be moved to a ClickUp task.

File: core/src/com/marama/view/View.java [hunk 2]
```diff
@@ -4,15 +4,35 @@
...
+        } catch (ExecuteException ee) {
+            ee.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        if (exitValue != 0) {
+            System.err.println("Editor: error, returned " + exitValue);
+            // TODO better error logging, static error code to string function from editor
+        }
+
```
Comment (wilbo) [inline core/src/com/marama/view/View.java]: You've said it in the comment already. I think it is better to extract this into its own class.

**Review by wilbo — CHANGES_REQUESTED** [review]
I don't think that all the commons-exec packages are needed. Can you remove the ones that are unnecessary? It also be nice if you could move the altered code in View.java into its own class.

File: core/src/com/marama/view/View.java [hunk 3]
```diff
@@ -4,15 +4,35 @@
...
+        try {
+            exitValue = executor.execute(cmdLine);
+        } catch (ExecuteException ee) {
+            ee.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        if (exitValue != 0) {
+            System.err.println("Editor: error, returned " + exitValue);
+            // TODO better error logging, static error code to string function from editor
```
Comment (LuneCoding) [inline core/src/com/marama/view/View.java]: TODO

**Comment by jjkramok** [comment]
I have updated my branch, please review again.

File: documentation/TechnicalDesignDocument.tex [hunk 4]
```diff
@@ -134,19 +118,41 @@
...
-    Diagrams need textual explanation of context, interpretation and underlying design decisions and reasoning.\\
-
+    \label{sec:designchoices}
     It has been decided to resizes texture assets to dimensions in the power of two.
     This is to maintain backwards compatibility for OpenGL ES 1.0, to keep all functionality (some libGDX features are otherwise not supported) and it is a bit more memory efficient.\cite{libgdxpottex}
 
+    \subsection[Communication MVA-MEA]{Receiving command or information from MVA and MEA}
+    \label{subsec:comchoice}
+    Concurrency is essential to this task.
+    This way the user interface and the marama logic are separated and would not delay each other.
```
Comment (LuneCoding) [inline documentation/TechnicalDesignDocument.tex]: Are separated
Would not
Fix tenses!

Comment (LuneCoding) [reply documentation/TechnicalDesignDocument.tex]: Probably the best to simply say that they 'Do not delay' each other.

File: documentation/TechnicalDesignDocument.tex [hunk 5]
```diff
@@ -191,11 +197,33 @@
     The first BoundingBox that is intersected by the ray will be the selected object.
     Casting the rays from the camera origin and locating the first intersected BoundingBox will prevent that 3D objects in the background will be selected.
 
+    \subsubsection{View and Game Communication}
+    View and editor (game) communication has everything to do with the way the two applications will work together.
+    The view is built in Java using libGDX, a graphics library and the editor is built using Scala.
```
Comment (LuneCoding) [inline documentation/TechnicalDesignDocument.tex]: > The view is built in Java using libGDX, a graphics library and the editor is built using Scala.
The view is built in Java using libGDX, a graphics library, and the editor is built using Scala.

**Review by LuneCoding — CHANGES_REQUESTED** [review]

File: documentation/TechnicalDesignDocument.tex (hunk 4 above)
Comment (jjkramok) [reply documentation/TechnicalDesignDocument.tex]: Are you going to make the changes, or should I?
Agreed that 'do' is better than 'would' in this case.


>Are separated
>Would not
>Fix tenses!

Why is every first letter capitalized? Is that a mistake or do I fail to see the reason behind it?

File: documentation/TechnicalDesignDocument.tex (hunk 5 above)
Comment (jjkramok) [reply documentation/TechnicalDesignDocument.tex]: Same thing here. Are you going to commit the changes or should I?

File: documentation/TechnicalDesignDocument.tex (hunk 4 above)
Comment (LuneCoding) [reply documentation/TechnicalDesignDocument.tex]: You should feel free to.

And... No rEaL ReAsoN

**Comment by jjkramok** [comment]
Updated my PR again, please review at earliest convenience.

**Review by wilbo — APPROVED** [review]

**Review by wilbo — CHANGES_REQUESTED** [review]
I found a maven source for the commons exec dependecy: https://mvnrepository.com/artifact/org.apache.commons/commons-exec/1.3

You can add the library in the build.gradle file with this gradle command: 
```
compile group: "org.apache.commons", name: "commons-exec", version: "1.3"
```

File: documentation/TechnicalDesignDocument.tex [hunk 6]
```diff
@@ -134,19 +118,41 @@
...
+        Coupling\cite{stackjavaconc} & tight & loose \\
+        \hline
+    \end{tabular}\\
+    There is little difference in speed\cite{stackjavaconc} between the two approaches.
+    A process is a separate task as far as the OS is concerned, whilst a thread runs on the allocated time of the parent process (the main thread in most cases).\\
+    One other way to use Processes is by use of Commons Exec\cite{commonsexec} library.
+    This library was made because the default Java implementation is inadequate.
+    The library works almost the same as Java \texttt{ProcessBuilder}.
+    To conclude, there is little practical difference in the two options.
+    Therefore the most lightweight and generally easier to implement option, threads, will be used for concurrency between the to projects.
```
Comment (ThijsvanDam) [inline documentation/TechnicalDesignDocument.tex]: between the **two** projects

File: documentation/TechnicalDesignDocument.tex [hunk 7]
```diff
@@ -191,11 +197,33 @@
...
     Casting the rays from the camera origin and locating the first intersected BoundingBox will prevent that 3D objects in the background will be selected.
 
+    \subsubsection{View and Game Communication}
+    View and editor (game) communication has everything to do with the way the two applications will work together.
+    The view is built in Java using libGDX, a graphics library, and the editor is built using Scala.
+    Both the view and the editor are two different applications both with its own goal.
+    The editor contains the Marama logic and the view will visualize this logic.
+    The following user stories will enforce that both these applications will work together:
+    \begin{itemize}
+        \item{\href{\clickup{2v50y}}{\#2v50y} As a MGD or player I want to be able to read and parse answers from the MVA}
```
Comment (ThijsvanDam) [inline documentation/TechnicalDesignDocument.tex:207]: Do we define MVA and MEA in the docs? Is this name approved by Chide?

File: documentation/TechnicalDesignDocument.tex [hunk 8]
```diff
@@ -191,11 +197,33 @@
...
+        \item{\href{\clickup{2v50h}}{\#2v50h} As a MGD or player I want to be able to read and parse answers from the MEA}
+        \item{\href{\clickup{2v516}}{\#2v516} As a MGD or player I want to be able to send messages to the MVA}
+    \end{itemize}
+    The communication is currently done using a .jar file.
+    A .jar file is exported from the editor application and imported inside the view application.
+    This is currently done using {\href{https://github.com/sbt/sbt-assembly}{sbt-assembly}}.
+    sbt-assembly is a plugin for sbt projects that exports a .jar file with al its dependencies.
+    The .jar file that is exported can be seen as a library and contains al information about the classes that are made within the editor application.
+    The view can address these classes if it is in need of some Marama logic.
+    This could be when a m-block is linked together with another m-block.
```
Comment (ThijsvanDam) [inline documentation/TechnicalDesignDocument.tex]: The class isn't called M-block anymore. Right now it's called Maramafication. Would it be better if Maramafication changed to M-block?

**Review by LuneCoding — APPROVED** [review]

File: core/src/com/marama/view/View.java [hunk 9]
```diff
@@ -4,8 +4,11 @@
 import com.badlogic.gdx.Gdx;
 import com.badlogic.gdx.graphics.GL20;
 import com.badlogic.gdx.utils.viewport.ScreenViewport;
-import com.badlogic.gdx.utils.viewport.Viewport;
 import com.marama.view.screens.SplashScreen;
+import org.apache.commons.exec.CommandLine;
+import org.apache.commons.exec.DefaultExecutor;
+import org.apache.commons.exec.ExecuteException;
```
Comment (ThijsvanDam) [inline core/src/com/marama/view/View.java]: Are these unused imports now?

**Review by ThijsvanDam — CHANGES_REQUESTED** [review]

File: documentation/TechnicalDesignDocument.tex (hunk 7 above)
Comment (jjkramok) [reply documentation/TechnicalDesignDocument.tex:207]: I don't think we do. And no, not explicitly

Comment (jjkramok) [reply documentation/TechnicalDesignDocument.tex:207]: I made a separate glossary.tex that is included in the main document.

**Comment by jjkramok** [comment]
I resolved your task @wilbo 
I removed the library and added the compile line to the build.gradle

**Pull request closed without merging**